testing/matrix-sdk-integration-testing/src/tests/sliding_sync/room.rs

```diff
@@ -853,6 +854,97 @@ async fn test_delayed_decryption_latest_event() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_delayed_invite_response_and_sent_message_decryption() -> Result<()> {
+    let alice = TestClientBuilder::new("alice").use_sqlite().build().await?;
+    let bob = TestClientBuilder::new("bob").use_sqlite().build().await?;
+
+    let alice_sync_service = SyncService::builder(alice.clone()).build().await.unwrap();
+    alice_sync_service.start().await;
+
+    let bob_sync_service = SyncService::builder(bob.clone()).build().await.unwrap();
+    bob_sync_service.start().await;
+
+    // alice creates a room and invites bob.
+    let alice_room = alice
+        .create_room(assign!(CreateRoomRequest::new(), {
+            invite: vec![],
+            is_direct: true,
+            preset: Some(RoomPreset::PrivateChat),
+        }))
+        .await?;
+    alice_room.enable_encryption().await?;
+
+    // Initial message to make sure any lazy /members call is performed before the
+    // test actually starts
+    alice_room
+        .send(RoomMessageEventContent::text_plain("dummy message to make members call"))
+        .await?;
+
+    // Set up sliding sync for bob.
+    let sliding_bob = bob
+        .sliding_sync("main")?
+        .with_all_extensions()
+        .poll_timeout(Duration::from_secs(2))
+        .network_timeout(Duration::from_secs(2))
+        .add_list(
+            SlidingSyncList::builder("all")
+                .sync_mode(SlidingSyncMode::new_selective().add_range(0..=20)),
+        )
+        .build()
+        .await?;
+
+    let s = sliding_bob.clone();
+    spawn(async move {
```

Comment:
nit: in general it's a better practice to `spawn` a task and keep the `join_handle` around, so that we can `.abort()` or `.await` on it later. In this case, we should abort at the end of the test.

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/3672/commits/f6a38134d60e241d04b0335d46cf45dbbe641554